include/seqan3/alphabet/structure/concept.hpp

```diff
@@ -0,0 +1,481 @@
+
+namespace seqan3
+{
+/*!\interface seqan3::RnaStructureAlphabet
+ * \brief A concept that indicates whether an alphabet represents RNA structure.
+ * \implements Alphabet
+ * \tparam structure_type The structure alphabet type.
+ * \ingroup structure
+ * \details RNA structure alphabets are required to represent interactions among RNA nucleotides.
+   Therefore, each structure letter can be categorised as unpaired, opening an interaction, or closing an interaction.
+   Additionally, the ability of representing pseudoknots is a property of RNA structure types.
+ */
+/*!\fn bool is_pair_open(structure_type const alph)
+ * \brief Check whether the given character represents a rightward interaction in an RNA structure.
+ * \relates seqan3::RnaStructureAlphabet
+ * \param alph The alphabet letter which is checked for the pairing property.
+ * \returns True if the letter represents a rightward interaction, False otherwise.
+ */
+/*!\fn bool is_pair_close(structure_type const alph)
+ * \brief Check whether the given character represents a leftward interaction in an RNA structure.
+ * \relates seqan3::RnaStructureAlphabet
+ * \param alph The alphabet letter which is checked for the pairing property.
+ * \returns True if the letter represents a leftward interaction, False otherwise.
+ */
+/*!\fn bool is_unpaired(structure_type const alph)
+ * \brief Check whether the given character represents an unpaired position in an RNA structure.
+ * \relates seqan3::RnaStructureAlphabet
+ * \param alph The alphabet letter which is checked for the pairing property.
+ * \returns True if the letter represents an unpaired site, False otherwise.
+ */
+/*!\fn std::optional<uint8_t> pseudoknot_id(structure_type const alph)
+ * \brief Get an identifier for a pseudoknotted interaction.
+ * \relates seqan3::RnaStructureAlphabet
+ * \param alph The alphabet letter which is checked for the pseudoknot id.
+ * \returns The pseudoknot id, if alph represents an interaction, and no value otherwise.
+ * It is guaranteed to be smaller than seqan3::max_pseudoknot_depth.
+ */
+/*!\var max_pseudoknot_depth<structure_type>
+ * \brief The ability of this alphabet to represent pseudoknots, i.e. crossing interactions, up to a certain depth.
+ * \relates seqan3::RnaStructureAlphabet
+ * \details
+ * It is the number of distinct pairs of interaction symbols the format supports. The value 1 denotes no
+ * pseudoknot support; any higher number gives the maximum nestedness. Value 0 is not allowed.
+ */
+//!\cond
+template <typename structure_type>
+SEQAN3_CONCEPT RnaStructureAlphabet = seqan3::Alphabet<structure_type> && requires(structure_type val)
+{
+    { seqan3::is_pair_open(val) } -> bool;
+    { seqan3::is_pair_close(val) } -> bool;
+    { seqan3::is_unpaired(val) } -> bool;
+    { seqan3::pseudoknot_id(val) } -> std::optional<uint8_t>;
+
+    // this is delegated to a static class variable, which must not be 0
+    requires seqan3::max_pseudoknot_depth<structure_type> > 0;
+};
```

Comment:
The same things  apply to this file: please add `seqan3::adaptation::` overload checks; simplify the concept definition and update the documentation!